packages/headless/src/controllers/product-listing/category-facet/headless-product-listing-category-facet.ts

```diff
@@ -0,0 +1,129 @@
+import {
+  logFacetUpdateSort,
+  logFacetShowMore,
+  logFacetShowLess,
+  logFacetClearAll,
+} from '../../../features/facets/facet-set/facet-set-analytics-actions';
+import {CategoryFacetSortCriterion} from '../../../features/facets/category-facet-set/interfaces/request';
+import {
+  CategoryFacetSearchSection,
+  CategoryFacetSection,
+  ConfigurationSection,
+  ProductListingSection,
+} from '../../../state/state-sections';
+import {getAnalyticsActionForCategoryFacetToggleSelect} from '../../../features/facets/category-facet-set/category-facet-set-controller-actions';
+import {
+  CategoryFacetOptions,
+  CategoryFacetSearchOptions,
+} from '../../core/facets/category-facet/headless-core-category-facet-options';
+import {determineFacetId} from '../../core/facets/_common/facet-id-determinor';
+import {CategoryFacetValue} from '../../../features/facets/category-facet-set/interfaces/response';
+import {
+  categoryFacetSearchSet,
+  categoryFacetSet,
+  configuration,
+  productListing,
+} from '../../../app/reducers';
+import {loadReducerError} from '../../../utils/errors';
+import {
+  buildCoreCategoryFacet,
+  CategoryFacet,
+  CategoryFacetProps,
+  CategoryFacetState,
+  CategoryFacetSearch,
+  CategoryFacetSearchState,
+  CategoryFacetSearchResult,
+} from '../../core/facets/category-facet/headless-core-category-facet';
+import {ProductListingEngine} from '../../../product-listing.index';
+import {fetchProductListing} from '../../../features/product-listing/product-listing-actions';
+
+export {
+  CategoryFacetValue,
+  CategoryFacetOptions,
+  CategoryFacetSearchOptions,
+  CategoryFacetProps,
+  CategoryFacet,
+  CategoryFacetState,
+  CategoryFacetSearch,
+  CategoryFacetSearchState,
+  CategoryFacetSearchResult,
+};
+
+/**
+ * Creates a `CategoryFacet` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @param props - The configurable `CategoryFacet` properties.
+ * @returns A `CategoryFacet` controller instance.
+ * */
+export function buildCategoryFacet(
+  engine: ProductListingEngine,
+  props: CategoryFacetProps
+): CategoryFacet {
+  if (!loadCategoryFacetReducers(engine)) {
+    throw loadReducerError;
+  }
+
+  const controller = buildCoreCategoryFacet(engine, props);
```

Comment:
Suggestion same as above:
```suggestion
  const coreController = buildCoreCategoryFacet(engine, props);
```